src/main/java/gift/repository/ProductRepositoryImpl.java

```diff
@@ -1,41 +1,84 @@
 package gift.repository;
 
 import gift.entity.Product;
-import java.util.ArrayList;
-import java.util.HashMap;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public class ProductRepositoryImpl implements ProductRepository {
 
-    private final Map<Long, Product> products = new HashMap<>();
+    private final JdbcTemplate jdbcTemplate;
 
-    private Long idNum = 1L;
+    public ProductRepositoryImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
 
     @Override
     public Product save(Product product) {
-        if (product.getId() == null) {
-            product.setId(idNum++);
+
+        String sql = "insert into product (name, imageUrl) values (?, ?)";
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            ps.setString(1, product.getName());
+            ps.setString(2, product.getImageUrl());
+            return ps;
+        }, keyHolder);
+
+        Number key = keyHolder.getKey();
+
+        if(key != null) {
+            product.setId(key.longValue());
         }
```

Comment:
지금의 코드도 흠잡을 곳이 없지만 아래와 같은 방식도 가능합니다ㅎㅎ
```suggestion
        Optional.ofNullable(keyHolder.getKey())
                .ifPresent(key -> product.setId(key.longValue()));
```